core/testcontainers/core/container.py

```diff
@@ -1,13 +1,18 @@
-import contextlib
-from platform import system
-from typing import Optional
+from __future__ import annotations
```

Comment:
are we sure this is what we want to do here?

Reply:
for the record, the `Optional[T]` is back in place, I've made `ruff` respect the runtime types.

Reply:
Replaced the `__future__` import with quoted type hints now as well. Easy to revert back if we want to, but as you mentioned on Slack it should probably be done project-wide if we want to go for that approach.